core/src/wallet/bitcoin/explorers/BitcoinLikeBlockchainExplorer.hpp

```diff
@@ -34,84 +34,78 @@
 #include <string>
 #include <chrono>
 #include <vector>
+
+#include <api/Error.hpp>
+#include <async/Future.hpp>
+#include <collections/collections.hpp>
+#include <math/BigInt.h>
 #include <utils/ConfigurationMatchable.h>
-#include "../../../utils/optional.hpp"
-#include "../../../api/ErrorCode.hpp"
-#include "../../../utils/Option.hpp"
-#include "../../../async/Future.hpp"
-#include "../../../collections/collections.hpp"
-#include "../../../math/BigInt.h"
-#include <wallet/common/Block.h>
+#include <utils/optional.hpp>
+#include <utils/Option.hpp>
+#include <wallet/common/explorers/AbstractBlockchainExplorer.h>
 
 namespace ledger {
     namespace core {
 
-        class BitcoinLikeBlockchainExplorer : public ConfigurationMatchable {
-        public:
-            typedef ledger::core::Block Block;
-
-            struct Input {
-                uint64_t index;
-                Option<BigInt> value;
-                Option<std::string> previousTxHash;
-                Option<uint32_t> previousTxOutputIndex;
-                Option<std::string> address;
-                Option<std::string> signatureScript;
-                Option<std::string> coinbase;
-                uint32_t sequence;
-                Input() {
-                    sequence = 0xFFFFFFFF;
-                };
-            };
-
-            struct Output {
-                uint64_t index;
-                std::string transactionHash;
-                BigInt value;
-                Option<std::string> address;
-                std::string script;
-                Output() = default;
-                std::string time;
+        struct BitcoinLikeBlockchainExplorerInput {
```

Comment:
Why moving them out of `BitcoinLikeBlockchainExplorer`?

Reply:
If I recall `BitcoinLikeBlockchainExplorer` uses `BitcoinLikeBlockchainExplorerTransaction` in template specialization